std/file.d

```diff
@@ -3932,6 +3932,8 @@ else version (Posix)
             _didLStat = false;
             _didStat = false;
             _dTypeSet = false;
+
+            _doStat();
```

Comment:
This completely subverts the intent of the original design and will cause a significant performance regression.

Reply:
Wow, thanks for the quick review! I was afraid that it wouldn’t be so easy, but after it passed the unit tests I was hopeful. I admit, I have hardly used stat, so I’m not familiar with all it’s intricacies.